docs/17-porting-tips.md

```diff
@@ -0,0 +1,333 @@
+# Porting tips
+
+## From 3.5 to 4.0
+
+The
+**best advice is to start from the default `rc.lua` and backport any changes**
+you might have added to your `rc.lua`.
+This avoids most of the possible errors due to missing important changes.
+
+If you still wish to ignore this advice, first read the
+<a href="./89-NEWS.md.html">NEWS</a> section about the breaking changes. This
+document assumes you did.
+
+Be warned, even if it looks like it's working after changing some lines, **it wont
+supports dynamic screen changes and will have many subtle bugs**. The changes
+mentioned below **are important for the stability of your window manager**.
+
+Here is a diff of the 3.5.9 `rc.lua` with the 4.0 one. All changes due to
+new features and new syntaxes have been removed. A `-` in front of the line
+correspond to content of the 3.5 `rc.lua` and `+` its replacement in 4.0.
+
+To test during the port, we recommend the `Xephyr` X11 server:
+
+    Xephyr :1 -screen 1280x800 &
+    DISPLAY=:1 path/to/awesome -c /path/to/rc.lua --search /path/to/awesome/lib
```

Comment:
Why the `--search`? I am not sure which path I am supposed to use here and I bet most users would have similar problems. Perhaps it makes sense to assume a prefix (e.g. `/usr/local`) and then give the full command based on that? When this is mentioned in the text, users should have a chance to adapt to their local paths.